src/util/mpz.cpp

```diff
@@ -435,6 +435,57 @@ void mpz_manager<SYNCH>::set(mpz & a, char const * val) {
     if (sign)
         neg(a);
 }
+template<bool SYNCH>
+void mpz_manager<SYNCH>::set_digits_unsigned(mpz & target, unsigned sz, unsigned const * digits) {
+
+  int n = 1;
+  bool little_endian = *(char *)&n == 1;
+
+  static_assert(sizeof(digit_t) >= sizeof(unsigned));
+  static_assert(sizeof(digit_t) % sizeof(unsigned) == 0);
+  auto us_per_digit = sizeof(digit_t) / sizeof(unsigned);
+  auto new_sz = sz % us_per_digit == 0 ? sz / us_per_digit : sz / us_per_digit + 1;
+
+  if (std::is_same<digit_t, unsigned>::value) {
+    set_digits(target, sz, digits);
+
+  } else if (little_endian && sz % us_per_digit == 0) {
+    set_digits(target, new_sz, (digit_t const*) digits);
+
+  } else {
+    // we need to allocate a new buffer
+    auto new_digits = new digit_t[new_sz];
```

Comment:
use unique_ptr please:
`auto new_digits = std::make_unique<digit_t[]>(new_sz);`

Then the delete below can go away.